src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int defaultCapacity = 16;
+    public static final double loadFactor = 0.75;
+    public static final int multiplier = 2;
+    private int size;
+    private Node<K,V>[] table = new Node[defaultCapacity];
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= (table.length * loadFactor)) {
+            resize();
+        }
+        int bucket = getBucket(key);
+        Node<K,V> lastNode = table[bucket];
+        if (lastNode == null) {
+            table[bucket] = new Node<>(key, value, null);
+            size++;
+        } else {
+            while (lastNode != null) {
+                if (Objects.equals(lastNode.key, key)) {
+                    lastNode.value = value;
+                    return;
+                } else if (lastNode.next == null) {
+                    lastNode.next = new Node<>(key, value, null);
+                    size++;
+                    return;
+                } else {
+                    lastNode = lastNode.next;
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int bucket = getBucket(key);
+        Node<K,V> node = table[bucket];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void resize() {
+        size = 0;
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length * multiplier];
+        for (Node<K, V> old : oldTable) {
+            while (old != null) {
+                this.put(old.key, old.value);
+                old = old.next;
+            }
+        }
+    }
```

Comment:
In the resize method, setting size to 0 and then using put() to reinsert elements can be inefficient. Consider directly transferring nodes to the new table without resetting the size and using put(), which recalculates the bucket and hash unnecessarily.